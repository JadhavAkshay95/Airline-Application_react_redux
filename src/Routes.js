import React from 'react';
import { createBrowserHistory } from 'history';
import { Router, Switch, Route } from 'react-router-dom';
import { Suspense, lazy } from 'react';

const AirlineStaffHomePage = React.lazy(() =>
  import('./component/airlineStaff/AirlineStaffHomePage')
);

const FlightDetails = lazy(() =>
  import('./component/airlineStaff/checkIn/flightDetails/FlightDetails')
);
const Login = lazy(() => import('./component/login/Login'));
const PassengerList = lazy(() =>
  import('./component/admin/passenger/PassengerList')
);

const FlightList = lazy(() =>
  import('./component/airlineStaff/checkIn/flightDetails/FlightList')
);

const Header = lazy(() => import('./component/common/Header'));
const history = createBrowserHistory();

const Routes = () => {
  return (
    <Suspense fallback={<div>Loading...</div>}>
      <Router history={history}>
        <Header history={history}></Header>
        <Switch>
          <Route exact path="/" component={Login}></Route>
          <Route
            exact
            path="/staff-home"
            component={AirlineStaffHomePage}
          ></Route>
          <Route
            exact
            path="/staff-home/in-flight/flight-list/:id"
            component={FlightDetails}
          ></Route>
          <Route
            exact
            path="/staff-home/check-in/flight-list/:id"
            component={FlightDetails}
          ></Route>
          <Route
            exact
            path="/admin/flight-list/:id"
            component={FlightDetails}
          ></Route>
          <Route exact path="/admin/flight-list" component={FlightList}></Route>
          <Route
            exact
            path="/admin/flight-list/:id/passengerList"
            component={PassengerList}
          ></Route>
          <Route
            exact
            path="/staff-home/check-in/flight-list"
            component={FlightList}
          ></Route>
          <Route
            exact
            path="/staff-home/in-flight/flight-list"
            component={FlightList}
          ></Route>
          <Route
            exact
            path="/staff-home/in-flight/flight-list/:id/passengerList"
            component={PassengerList}
          ></Route>
          <Route
            exact
            path="/staff-home/check-in/flight-list/:id/passengerList"
            component={PassengerList}
          ></Route>
        </Switch>
      </Router>{' '}
    </Suspense>
  );
};

export default Routes;
