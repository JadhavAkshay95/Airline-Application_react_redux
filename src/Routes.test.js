import React from 'react';
import { shallow } from 'enzyme';
import Routes from './Routes';

const renderRoutes = args => {
  const defaultProps = {};
  const props = { ...defaultProps, ...args };
  return shallow(<Routes {...props} />);
};

describe('Render manage service component', () => {
  it('Render manage service modal', () => {
    let wrapper = renderRoutes({});
    expect(wrapper).toMatchSnapshot();
  });
});
