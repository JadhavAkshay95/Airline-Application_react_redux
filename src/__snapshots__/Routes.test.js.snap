// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render manage service component Render manage service modal 1`] = `ShallowWrapper {}`;
